infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/RestApiRequest.java

```diff
@@ -50,6 +50,17 @@ public void respondOk(final Object response) throws JsonProcessingException {
     respond(HttpStatusCodes.SC_OK, JSON_CONTENT_TYPE, response);
   }
 
+  public void respondAsync(final SafeFuture<AsyncApiResponse> futureResponse) {
+    context.future(
+        futureResponse.thenAccept(
+            result -> {
+              context.status(result.getResponseCode());
+              if (result.hasResponseBody()) {
+                context.json(result.getResponseBody());
+              }
```

Comment:
Would be simpler if `result.getResponseBody()` just returned an `Optional` then you could use `.ifPresent`.